.github/workflows/function_test_defender.yaml

```diff
@@ -0,0 +1,572 @@
+# Run functional tests for Defender
+
+name: Function Test Defender
+
+# This is a reusable workflow called by the pipeline.
+on:
+  workflow_call:
+    inputs:
+      EmailOnFailure:
+        description: "The email address to send if workflow fails."
+        required: true
+        type: string
+      GitHubEventSchedule:
+        description: "A type of GitHub event"
+        required: true
+        type: string
+      GitHubEventName:
+        description: "The name of a GitHub event"
+        required: true
+        type: string
+    secrets:
+      PfxBase64:
+        required: true
+      PfxPassword:
+        required: true
+      TestParams:
+        required: true
+      TestParamsDev:
+        required: true
+      NotifierUsername:
+        required: true
+      NotifierPassword:
+        required: true
+  workflow_dispatch:
+
+permissions:
+  id-token: write
+  contents: write
+
+jobs:
+  Def_Tenant1:
+    name: Def_Tenant1
+    runs-on: windows-latest
+    environment: Development
+    permissions:
+      id-token: write
+      contents: write
+    concurrency:
+      group: Defender_Tenant1
+      cancel-in-progress: false
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+        with:
+          path: repo
+      - name: Import PFX
+        uses: ./repo/.github/actions/import-pfx
+        with:
+          PfxBase64: ${{ secrets.PfxBase64 }}
+          PfxPassword: ${{ secrets.PfxPassword }}
+      - name: Get Thumbprint
+        id: get-thumbprint
+        uses: ./repo/.github/actions/get-thumbprint
+        with:
+          PfxPassword: ${{ secrets.PfxPassword }}
+      - name: Restore ScubaGear
+        uses: actions/cache/restore@v4
+        with:
+          path: C:\Program Files\WindowsPowerShell\Modules\ScubaGear
+          key: scubagear-directory-
+      - name: Restore Dependencies
+        uses: actions/cache/restore@v4
+        with:
+          path: C:\Users\runneradmin\Documents\WindowsPowerShell\Modules
+          key: powershell-directory-
+      - name: Restore OPA
+        uses: actions/cache/restore@v4
+        with:
+          path: C:\Users\runneradmin\.scubagear\Tools
+          key: opa-directory-
```

Comment:
If Publish fails, then the workflow itself fails and the test jobs do not run.

The caches are no longer always deleted, so there should not be any problem re-running jobs.

Thus, the only way this is an issue is if Publish succeeds and someone manages to delete those caches before for the test job runs.  This would have to be a very intentional action on someone's part, not something that could occur normally.  While this is possible, I think this is an extreme edge case that is not a legit item of concern.  I vote for doing nothing or perhaps putting an item in the backlog to handle this edge case at a later date (when we don't have more important things to do).